### PR TITLE
[SPARK-10278][MLlib][PySpark] Add @since annotation to pyspark.mllib.tree

### PR DESCRIPTION
**Pull request by yu-iskw**

### CODE REVIEW

**Comment by yu-iskw** [comment]
Jenkins, retest this please.

File: python/pyspark/mllib/tree.py [hunk 1]
```diff
@@ -90,16 +101,23 @@ def predict(self, x):
...
+        """Get depth of tree.
+        E.g.: Depth 0 means 1 leaf node.  Depth 1 means 1 internal node and 2 leaf nodes.
+        """
         return self._java_model.depth()
 
     def __repr__(self):
         """ summary of model. """
         return self._java_model.toString()
 
+    @since("1.3.0")
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: This is from 1.2.0

File: python/pyspark/mllib/tree.py [hunk 2]
```diff
@@ -30,6 +30,11 @@
 
 
 class TreeEnsembleModel(JavaModelWrapper, JavaSaveable):
+    """TreeEnsembleModel
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/tree.py [hunk 3]
```diff
@@ -72,7 +80,10 @@ class DecisionTreeModel(JavaModelWrapper, JavaSaveable, JavaLoader):
     .. note:: Experimental
 
     A decision tree model for classification or regression.
+
+    .. addedversion:: 1.1.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/tree.py [hunk 4]
```diff
@@ -115,6 +133,8 @@ class DecisionTree(object):
 
     Learning algorithm for a decision tree model for classification or
     regression.
+
+    .. addedversion:: 1.1.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/tree.py [hunk 5]
```diff
@@ -239,6 +261,8 @@ class RandomForestModel(TreeEnsembleModel, JavaLoader):
     .. note:: Experimental
 
     Represents a random forest model.
+
+    .. addedversion:: 1.2.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/tree.py [hunk 6]
```diff
@@ -252,6 +276,8 @@ class RandomForest(object):
 
     Learning algorithm for a random forest model for classification or
     regression.
+
+    .. addedversion:: 1.2.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/tree.py [hunk 7]
```diff
@@ -418,6 +446,8 @@ class GradientBoostedTreesModel(TreeEnsembleModel, JavaLoader):
     .. note:: Experimental
 
     Represents a gradient-boosted tree model.
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/tree.py [hunk 8]
```diff
@@ -431,6 +461,8 @@ class GradientBoostedTrees(object):
 
     Learning algorithm for a gradient boosted trees model for
     classification or regression.
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/tree.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/util.py [hunk 9]
```diff
@@ -32,6 +32,8 @@ class MLUtils(object):
 
     """
     Helper methods to load, save and pre-process data used in MLlib.
+
+    .. addedversion:: 1.0.0
```
Comment (noel-smith) [inline python/pyspark/mllib/util.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/util.py [hunk 10]
```diff
@@ -197,6 +204,8 @@ def loadVectors(sc, path):
 class Saveable(object):
     """
     Mixin for models and transformers which may be saved as files.
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/util.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/util.py [hunk 11]
```diff
@@ -222,9 +231,13 @@ class JavaSaveable(Saveable):
     """
     Mixin for models that provide save() through their Scala
     implementation.
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/util.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/util.py [hunk 12]
```diff
@@ -235,6 +248,8 @@ def save(self, sc, path):
 class Loader(object):
     """
     Mixin for classes which can load saved models from files.
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/util.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/util.py [hunk 13]
```diff
@@ -256,6 +271,8 @@ class JavaLoader(Loader):
     """
     Mixin for classes which can load saved models using its Scala
     implementation.
+
+    .. addedversion:: 1.3.0
```
Comment (noel-smith) [inline python/pyspark/mllib/util.py]: Change to `.. versionadded::`

File: python/pyspark/mllib/util.py [hunk 14]
```diff
@@ -280,15 +297,21 @@ def _load_java(cls, sc, path):
...
+        """Load a model from the given path."""
         java_model = cls._load_java(sc, path)
         return cls(java_model)
 
 
 class LinearDataGenerator(object):
-    """Utils for generating linear data"""
+    """Utils for generating linear data.
+
+    .. addedversion:: 1.5.0
```
Comment (noel-smith) [inline python/pyspark/mllib/util.py]: Change to `.. versionadded::`

**Comment by noel-smith** [comment]
Just minor fixes needed - change `addedversion` to `versionadded` + one version number alteration.

**Comment by noel-smith** [comment]
Thanks for the changes - LGTM!

**Comment by yu-iskw** [comment]
@noel-smith thank you for reviewing it.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**